src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -781,8 +812,21 @@ public void prepareForTranslogRecovery() throws ElasticsearchException {
         // we disable deletes since we allow for operations to be executed against the shard while recovering
         // but we need to make sure we don't loose deletes until we are done recovering
         config.setEnableGcDeletes(false);
-        createNewEngine();
         recoveryState.setStage(RecoveryState.Stage.TRANSLOG);
+        createNewEngine(skipTranslogRecovery);
```

Comment:
The recovery state reporting is not clean now, as we start the engine and reply translog under translog. I wonder if it's worth while capturing the START phase at all (i.e. how it takes to open the lucene index). Maybe we should just rename that phase to CHECK_INDEX and only apply it if it's needed ? (i.e., `checkIndexOnStartup == true`)